flink-doris-connector/src/main/java/org/apache/doris/flink/serialization/RowBatch.java

```diff
@@ -153,6 +163,37 @@ public RowBatch readArrow() {
         }
     }
 
+    public RowBatch readFlightArrow() {
+        try {
+            this.root = arrowStreamReader.getVectorSchemaRoot();
+            fieldVectors = root.getFieldVectors();
+            if (fieldVectors.size() > schema.size()) {
+                logger.error(
+                        "Schema size '{}' is not equal to arrow field size '{}'.",
+                        fieldVectors.size(),
+                        schema.size());
+                throw new DorisException(
+                        "Load Doris data failed, schema size of fetch data is wrong.");
+            }
+            if (fieldVectors.isEmpty() || root.getRowCount() == 0) {
+                logger.debug("One batch in arrow has no data.");
+                return null;
+            }
+            rowCountInOneBatch = root.getRowCount();
+            for (int i = 0; i < rowCountInOneBatch; ++i) {
+                rowBatch.add(new RowBatch.Row(fieldVectors.size()));
+            }
+            convertArrowToRowBatch();
+            readRowCount += root.getRowCount();
+            return this;
+        } catch (DorisException e) {
+            logger.error("Read Doris Data failed because: ", e);
+            throw new DorisRuntimeException(e.getMessage());
+        } catch (IOException e) {
+            return this;
```

Comment:
Isn't it necessary to throw an exception here?

Reply:
Call the close method in finally

Reply:
In readFlightArrow, since the client does not have a hasNext check, it is necessary to loop and check if loadnextbatch has data. If closed in the finally method, it will result in not being able to retrieve data properly and in some cases cause an arrow memory leak. Close the client using the close method in DorisSourceSplitReader.